hera_qm/vis_metrics.py

```diff
@@ -48,6 +51,156 @@ def check_noise_variance(data):
     return Cij
 
 
+def sequential_diff(data, t_int=None, axis=(0,), pad=True, history=''):
+    """
+    Take a sequential (forward) difference of a visibility waterfall
+    as an estimate of the visibility noise. Note: the 1/sqrt(2) correction
+    needed after taking a difference is not applied to the data directly,
+    but is taken into account by the output t_int. If input data is a UVData
+    object, this correction is multiplied into the nsample_array.
+
+    Parameters
+    ----------
+    data : ndarray or UVData object
+        A 2D or 3D ndarray containing visibility data, with
+        shape (Ntimes, Nfreqs, :). Or UVData object
+        holding visibility data.
+
+    t_int : ndarray
+        A 2D or 3D ndarray containing the integration time
+        of the visibilities matching shape of data.
+        If None and data is UVData, will use UVData ~flags *
+        nsample * integration_time as input.
+
+    axis : int or tuple
+        Axes along which to take sequential difference
+
+    pad : boolean
+        If True, insert an extra (flagged) column at the end
+        of axis such that diff_data has same shape as input.
+
+    history : str
```

Comment:
I just noticed while reviewing your latest commits, that this argument isn't actually used in the function.

Reply:
ah thanks, fixed that